packages/react-native/React/Base/RCTBundleURLProvider.h

```diff
@@ -118,20 +119,23 @@ RCT_EXTERN void RCTBundleURLProviderAllowPackagerServerAccess(BOOL allowed);
  * - enableDev: Whether to keep or remove `__DEV__` blocks from the bundle.
  * - enableMinification: Enables or disables minification. Usually production bundles are minified and development
  *     bundles are not.
+ * - inlineSourceMap: Determines whether the source map is inline in the JS bundle. This is useful for debugging with Safari /JavascriptCore.
  * - modulesOnly: When true, will only send module definitions without polyfills and without the require-runtime.
  * - runModule: When true, will run the main module after defining all modules. This is used in the main bundle but not
  *     in split bundles.
  */
 + (NSURL *)jsBundleURLForBundleRoot:(NSString *)bundleRoot
                        packagerHost:(NSString *)packagerHost
                           enableDev:(BOOL)enableDev
-                 enableMinification:(BOOL)enableMinification;
+                 enableMinification:(BOOL)enableMinification
+                    inlineSourceMap:(BOOL)inlineSourceMap;
 
 + (NSURL *)jsBundleURLForBundleRoot:(NSString *)bundleRoot
                        packagerHost:(NSString *)packagerHost
                      packagerScheme:(NSString *)scheme
                           enableDev:(BOOL)enableDev
                  enableMinification:(BOOL)enableMinification
+                    inlineSourceMap:(BOOL)inlineSourceMap
                         modulesOnly:(BOOL)modulesOnly
                           runModule:(BOOL)runModule;
```

Comment:
These changes are breaking for all the users. They will all have to update their apps and I'm not really ok in doing this, in this way, tbh.

May I suggest that, instead of modifying the current `jsBundleURLForBundleRoot` methods, we follow this approach:

1. We add two new methods, with the new parameter.
2. We have the old methods calls the new ones with a default which reflect the old behavior (I assume that the default must be `NO`, right?)
3. We deprecate the old methods with the proper annotation.

In this way, users don't have to do anything when they update to 0.73. They will only start receiving some warning, but everything will continue working.
We can also point them to some documentation on how to `inlineSourceMap` if they need.

Then, in 0.74, we can decide to remove the deprecated methods. But, in this way, they will be warned about the change.

Reply:
Yeah, the point is that we can't really know how our users may use this in general. For example, Expo could use those APIs instead of the one we are using in RNTester. Or some companies may have their own private packager/app setup which use those APIs... So I'd rather be safe than sorry! ;)